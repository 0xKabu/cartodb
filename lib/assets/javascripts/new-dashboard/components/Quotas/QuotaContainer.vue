<template>
  <div class="quota-container">
    <slot></slot>
  </div>
</template>

<script>
export default {
  name: 'QuotaCard'
};
</script>

<style scoped lang="scss">
@import 'stylesheets/new-dashboard/variables';

.quota-container {
  height: 100%;
  padding: 24px;
  border: 1px solid $light-grey;
  border-radius: 2px;
  box-shadow: 1px 1px 1px 0 #EEE;
}
</style>
