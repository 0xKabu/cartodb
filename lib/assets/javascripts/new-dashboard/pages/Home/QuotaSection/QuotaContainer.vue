<template>
  <div class="quota-container">
    <div class="quota-headers">
      <div class="quota-cell cell--main">
        <h3 class="title is-caption quota-title">{{title}}</h3>
      </div>

      <div class="quota-labels">
        <div class="quota-cell cell--medium">
          <span class="text is-small is-txtSoftGrey">{{ $t('QuotaSection.header.remaining') }}</span>
        </div>
        <div class="quota-cell cell--medium cell--mobile">
          <span class="text is-small is-txtSoftGrey">{{ $t('QuotaSection.header.used') }}</span>
        </div>
        <div class="quota-cell cell--medium">
          <span class="text is-small is-txtSoftGrey quota-total" :class="{'is-active' : perMonth }">{{$t(`QuotaSection.header.total`)}}</span>
          <span class="text is-small is-txtSoftGrey quota-month" :class="{'is-active' : perMonth }">&nbsp;{{$t(`QuotaSection.header.perMonth`)}}</span>
        </div>
        <div class="quota-cell cell--small"></div>
      </div>
    </div>
    <div class="quota-info">
      <slot></slot>
    </div>
  </div>
</template>

<script>
export default {
  name: 'QuotaContainer',
  props: {
    title: String,
    perMonth: Boolean
  }
};
</script>

<style scoped lang="scss">
@import 'stylesheets/new-dashboard/variables';

.quota-info {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  background-color: $white;
}

.quota-title {
  margin-bottom: 8px;
}

.quota-headers {
  display: flex;
  justify-content: space-between;
}

.quota-labels {
  display: flex;
}

.quota-cell {
  display: flex;
  position: relative;
  flex-grow: 0;
  flex-shrink: 0;
  align-items: center;
  padding: 0 10px;
  text-overflow: ellipsis;
  white-space: nowrap;

  &:first-of-type {
    padding-left: 0;
  }

  &:last-of-type {
    padding-right: 0;
  }

  @media (max-width: $layout-tablet) {
    padding: 0 5px;
  }
}

.cell--mobile {
  @media (max-width: $layout-mobile) {
    display: none;
  }
}

.cell--main {
  flex-grow: 1;
  flex-shrink: 1;
  min-width: 300px;

  @media (max-width: $layout-mobile) {
    min-width: unset;
  }
}

.cell--medium {
  width: 120px;

  @media (max-width: $layout-tablet) {
    width: 100px;
  }
}

.cell--small {
  width: 58px;

  @media (max-width: $layout-tablet) {
    display: none;
  }
}

.quota-month {
  display: none;
}

.is-active {
  &.quota-total {
    @media (max-width: $layout-tablet) and (min-width: $layout-mobile) {
      &::after {
        content: '*';
      }
    }
  }

  &.quota-month {
    display: flex;

    @media (max-width: $layout-tablet) and (min-width: $layout-mobile) {
      display: none;
    }

    &::after {
      content: '*';
    }
  }
}

</style>
