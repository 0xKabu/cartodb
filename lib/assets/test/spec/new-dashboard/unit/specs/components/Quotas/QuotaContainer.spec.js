import { shallowMount } from '@vue/test-utils';
import QuotaContainer from 'new-dashboard/components/Quotas/QuotaContainer';

describe('QuotaContainer.vue', () => {
  it('should render correct contents', () => {
    const quotaContainer = shallowMount(QuotaContainer);

    expect(quotaContainer).toMatchSnapshot();
  });
});
