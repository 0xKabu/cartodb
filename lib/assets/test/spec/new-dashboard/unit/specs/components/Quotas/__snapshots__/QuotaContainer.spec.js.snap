// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuotaContainer.vue should render correct contents 1`] = `<div class="quota-container"></div>`;
